src/ir/eh-utils.h

```diff
@@ -40,6 +41,22 @@ void handleBlockNestedPop(Try* try_, Function* func, Module& wasm);
 // Calls handleBlockNestedPop for each 'Try's in a given function.
 void handleBlockNestedPops(Function* func, Module& wasm);
 
+// Given a catch body, find the pop corresponding to the catch. There might be
+// pops nested inside a try inside this catch, and we must ignore them, like
+// here:
+//
+//  (catch
+//    (pop) ;; we want this for the outer catch
+//    (try
+//      (catch
+//        (pop) ;; but we do not want this for the outer catch
+Pop* findPop(Expression* expr);
+
+// Like findPop(), but it does *not* assume that the module validates. A catch
+// might therefore have 0 or more than 1 pop. This is primarily useful in the
```

Comment:
Having 0 pop can be valid when a tag does not have params: https://github.com/WebAssembly/binaryen/blob/6257a5857a42fec6b86352b03c1e92dcb7a490ea/src/wasm/wasm-validator.cpp#L2199-L2204

This comment sounds like having 0 pop is invalid. Sorry I should've been more clear about this in my earlier comment.